### PR TITLE
UI/OIDC client form

### PR DESCRIPTION
**Pull request by hellobontempo**
Application (client) create view for OIDC provider config
create-app

### CODE REVIEW

File: ui/app/templates/components/oidc/assignment-form.hbs
Comment (hashishaw) [inline ui/app/templates/components/oidc/assignment-form.hbs:1]: This comment is complete, yes?

File: ui/app/templates/components/oidc/client-form.hbs
Comment (hashishaw) [inline ui/app/templates/components/oidc/client-form.hbs:63]: This seems like something superfluous we could assume in the SearchSelectWithModal template and remove from the params

File: ui/lib/core/addon/components/form-field-groups.hbs
Comment (hashishaw) [inline ui/lib/core/addon/components/form-field-groups.hbs:7]: Thanks for cleaning this up 😵 
`if (!(mode !== "create" && attr.name === "name"))`

I wonder if we should call out `isNew` on the component model as something that's only present on models from the search select 🤔 Not sure we have any documentation about what this component expects from the model so that might not be worth doing on this PR. A potential motivation to move some of these to typescript though 👀

File: ui/lib/core/addon/components/search-select-with-modal.js
Comment (hashishaw) [inline ui/lib/core/addon/components/search-select-with-modal.js:46]: 😍 ✨

File: ui/lib/core/addon/utils/search-select-has-many.js
Comment (hashishaw) [inline ui/lib/core/addon/utils/search-select-has-many.js:30]: Since we're passing in `model` as a param, I would change either the param name or this and line 34. This line is actually safe as a let within the `if` but line 34 is returning to the function param `model` so this may not behave as expected. 
Also, let's make sure we have good test coverage for this util

**Review by hashishaw — COMMENTED** [review]
Great work so far!

File: ui/app/templates/components/oidc/assignment-form.hbs
Comment (hellobontempo) [reply ui/app/templates/components/oidc/assignment-form.hbs:1]: I added it just for show when recording the gif - but was waiting until Angel had merged her PR before actually making the change!

File: ui/lib/core/addon/components/search-select-with-modal.js
Comment (hellobontempo) [reply ui/lib/core/addon/components/search-select-with-modal.js:60]: Thanks for calling this out! I glimmerized this based on the original `search-select` component where this is a computed property:
```  
inputValue: computed(function () {
    return [];
  }),
  ```
  
But I didn't/don't fully understand the purpose of the original computed property and wasn't sure if it was something the EmberPowerSelect needed?

File: ui/lib/core/addon/components/search-select-with-modal.js
Comment (hellobontempo) [reply ui/lib/core/addon/components/search-select-with-modal.js:68]: This is a great point! Good call

File: ui/lib/core/addon/components/search-select-with-modal.js
Comment (hellobontempo) [reply ui/lib/core/addon/components/search-select-with-modal.js:96]: Yep! Realized this when I removed the getter - thank you!

File: ui/lib/core/addon/components/search-select.js
Comment (hellobontempo) [reply ui/lib/core/addon/components/search-select.js:32]: good idea! i really like verbiage feed back 😍

File: ui/lib/core/addon/utils/search-select-has-many.js
Comment (hellobontempo) [reply ui/lib/core/addon/utils/search-select-has-many.js:30]: yes - agreed on test coverage! 

i was having issues with the model saving, but through _lots_ of trial, I think I fixed the issue and was able to remove model as a param (which I didn't totally love doing in the first place).

File: ui/app/templates/components/oidc/assignment-form.hbs
Comment (hellobontempo) [inline ui/app/templates/components/oidc/assignment-form.hbs:29]: This large diff is just changes in indentation from adding the `{{#unless}}` block and removing the `div`

**Review by Monkeychip — APPROVED** [review]
Amazing work on this!

**Pull request merged**
